aiida/cmdline/params/types/identifier.py

```diff
@@ -46,24 +47,29 @@ def __init__(self, sub_classes=None):
             will be mapped upon. These classes have to be strict sub classes of the base orm class defined
             by the orm class loader
         """
-        from aiida.common import exceptions
+        if sub_classes is not None and not isinstance(sub_classes, tuple):
+            raise TypeError('sub_classes should be a tuple of entry point strings')
 
         self._sub_classes = None
-        self._entry_points = []
-
-        if sub_classes is not None:
+        self._entry_point_strings = sub_classes
 
-            if not isinstance(sub_classes, tuple):
-                raise TypeError('sub_classes should be a tuple of entry point strings')
+    @cached_property
```

Comment:
I am not sure how `cached_property` is implemented, but wouldn't it be easier to implement the "caching" ourselves here. E.g.:
```python
    def __init__(self, sub_classes=None):
        if sub_classes is not None and not isinstance(sub_classes, tuple):
            raise TypeError('sub_classes should be a tuple of entry point strings')

        self._sub_classes = None
        self._entry_point_strings = sub_classes
        self._entry_points = None

    def entry_points(self):
        """Return list of allowed entry points."""
        from aiida.common import exceptions

        if self._entry_points is not None:
            return self._entry_points

        if self._entry_point_strings is None:
            return []

        entry_points = []

        for entry_point_string in self._entry_point_strings:
            try:
                entry_point = get_entry_point_from_string(entry_point_string)
            except (ValueError, exceptions.EntryPointError) as exception:
                raise ValueError(f'{entry_point_string} is not a valid entry point string: {exception}')
            else:
                entry_points.append(entry_point)
        
        self._entry_points = entry_points

        return self._entry_points
```
i.e. just make `entry_points` a public property and have `_entry_points` be `None` initially and populate it the first time `entry_points` is called.

Reply:
I guess we have a different definition of easier :smile: To me the  decorator seems clearer, it's one less level of indirection, and allowed me to not change the code elsewhere (i.e. using the `entry_points` instead of `_entry_points`). But I don't have a strong feeling if you think I should change it.

Reply:
I shouldn't have said easier. I was mostly wondering how `cached_property` implements this under the hood. Will it be as efficient as the "manual" version? if there is no (noticable) difference, then I agree that using the cached_property is easier to read and fine to keep that.

Reply:
Sure, I'll do a microbenchmark and report back. I would expect this to make much difference because I don't think this property should be accessed too many times.

Reply:
Please, don't spend too much time on it. It is not that important, we can keep the `cached_property` approach

Reply:
ok, I may do it later and we can revisit this later, but don't want to block this PR on this.